### ISSUE TITLE
Failing to find the root entities

### ISSUE THREAD

**Issue by starksm64** [jakartaredhat/hibernate-data-tck#4]
With the latest tck runner setup the repository beans are being injected, and hibernate is configured, but I'm failing to see the tck entity tables being created, and when I run one the `StandaloneEntityTests#ensureNaturalNumberPrepopulation` test from the command line, there is no 

```bash
starksm@Scotts-Mac-Studio runner % mvn test -Dtest=org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests#ensureNaturalNumberPrepopulation
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------< org.hibernate:hiberate-data-tck-runner >---------------
[INFO] Building Hibernate Jakarta Data TCK Runner 1.0
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ hiberate-data-tck-runner ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/starksm/Dev/Jakarta/hibernate-data-tck/runner/src/main/resources
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ hiberate-data-tck-runner ---
[INFO] No sources to compile
[INFO] 
[INFO] --- resources:3.3.1:testResources (default-testResources) @ hiberate-data-tck-runner ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 4 resources from src/test/resources to target/test-classes
[INFO] 
[INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ hiberate-data-tck-runner ---
[INFO] Nothing to compile - all classes are up to date.
[INFO] 
[INFO] --- surefire:2.22.2:test (default-test) @ hiberate-data-tck-runner ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests
HibernateStandaloneExtension.beforeAll
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.997 s <<< FAILURE! - in org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests
[ERROR] ensureNaturalNumberPrepopulation  Time elapsed: 0.983 s  <<< ERROR!
jakarta.data.exceptions.DataException: org.hibernate.query.sqm.UnknownEntityException: Could not resolve root entity 'NaturalNumber'
Caused by: org.hibernate.query.sqm.UnknownEntityException: Could not resolve root entity 'NaturalNumber'
```

This is coming from the test calling the numbers.countBy() method, which is hitting the following `NaturalNumbers_#countBy()` generated method:

```java
package ee.jakarta.tck.data.framework.read.only;
...
@RequestScoped
@Generated("org.hibernate.processor.HibernateProcessor")
public class NaturalNumbers_ implements NaturalNumbers {
...
	@Override
	public long countBy() {
		try {
			return session.createSelectionQuery(COUNT_BY, long.class)
				.getSingleResult();
		}
		catch (NoResultException exception) {
			throw new EmptyResultException(exception);
		}
		catch (NonUniqueResultException exception) {
			throw new jakarta.data.exceptions.NonUniqueResultException(exception);
		}
		catch (PersistenceException exception) {
			throw new DataException(exception);
		}
	}
```

The tck run.log which has debugging cranked up is attached:
[DataTCK00.log](https://github.com/jakartaredhat/hibernate-data-tck/files/14814545/DataTCK00.log)

**Comment by gavinking** [comment]
I don't get this, I get:

```
[INFO] Running org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests
HibernateStandaloneExtension.beforeAll
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.862 s <<< FAILURE! - in org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests
[ERROR] ensureNaturalNumberPrepopulation  Time elapsed: 0.809 s  <<< ERROR!
org.jboss.weld.exceptions.DeploymentException: 
WELD-001408: Unsatisfied dependencies for type MultipleEntityRepo with qualifiers @Default
  at injection point [BackedAnnotatedField] @Inject ee.jakarta.tck.data.standalone.entity.EntityTests.shared
  at ee.jakarta.tck.data.standalone.entity.EntityTests.shared(EntityTests.java:0)
```

**Comment by gavinking** [comment]
Also, I'm not really clear on how Hibernate is being bootstrapped here. Something needs to be creating an `EntityManagerFactory` that the repository implementations can inject using:

```java
@PersistenceUnit
private EntityManagerFactory sessionFactory;
```

But I don't see where that's happening.

**Comment by starksm64** [comment]
The org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests class in runner/src/test/java/org/hibernate/ee/jakarta/data/tck/StandaloneEntityTests.java is setting up Weld and Hibernate:

```java
@ExtendWith(WeldJunit5Extension.class)
public class StandaloneEntityTests extends EntityTests {
    static class EMFAction implements Function<InjectionPoint, EntityManagerFactory> {
        @Override
        public EntityManagerFactory apply(InjectionPoint injectionPoint) {
            HashMap<String, Object> properties = new HashMap<>();
            BeanManager beanManager = CDI.current().getBeanManager();
            System.out.println("BeanManager: " + beanManager);
            properties.put("jakarta.persistence.bean.manager", beanManager);
            properties.put("javax.persistence.bean.manager", beanManager);
            return Persistence.createEntityManagerFactory("jakarta-data-tck", properties);
        }
    }
    @WeldSetup
    public WeldInitiator weld = WeldInitiator.from(EntityTests.class,
                    AsciiCharacters_.class,
                    Boxes_.class,
                    NaturalNumbers_.class,
                    PositiveIntegers_.class,
                    CustomRepository_.class,
                    DirectoryBuildCompatibleExtension.class)
            .activate(RequestScoped.class)
            .setPersistenceUnitFactory((Function) new EMFAction())
            .build()
            ;

}
```

**Comment by starksm64** [comment]
> [ERROR] ensureNaturalNumberPrepopulation  Time elapsed: 0.809 s  <<< ERROR!
org.jboss.weld.exceptions.DeploymentException: 
WELD-001408: Unsatisfied dependencies for type MultipleEntityRepo with qualifiers @Default
  at injection point [BackedAnnotatedField] @Inject ee.jakarta.tck.data.standalone.entity.EntityTests.shared
  at ee.jakarta.tck.data.standalone.entity.EntityTests.shared(EntityTests.java:0)

I don't see where a `shared` field is defined, and I also don't see a `MultipleEntityRepo` type anywhere.

**Comment by gavinking** [comment]
> I don't see where a `shared` field is defined, and I also don't see a `MultipleEntityRepo` type anywhere.

I guess you need to update and rebuild the TCK.

**Comment by gavinking** [comment]
> The `org.hibernate.ee.jakarta.data.tck.StandaloneEntityTests` class ...  is setting up Weld and Hibernate

OK, but that's just using `Persistence.createEntityManagerFactory()`, and `persistence.xml` doesn't have a list of entity classes. I think they need to be listed explicitly, don't they. (We're not running in an EE container here.)

**Comment by gavinking** [comment]
My #5 fixes this problem, and a couple of small fixes to `hibernate-orm` gets me as far as this:

<img width="871" alt="Screenshot 2024-03-31 at 7 46 48 PM" src="https://github.com/jakartaredhat/hibernate-data-tck/assets/579974/79ae4afa-6f50-4301-93f8-b9061d9a2743">

That is, 60/73 tests successful.

Scanning them, the test failures are probably mostly minor bugs on our side, to do with pagination and sorting.

You'll need to pull and rebuild `hibernate-jpamodelgen`.

**Comment by gavinking** [comment]
> Scanning them, the test failures are probably mostly minor bugs on our side, to do with pagination and sorting.

This seems to have been a bad guess.

I'm down to 9 failures, and the ones I've investigated don't appear to be the result of bugs, but rather due to side-effects of creating a new EMF for each repository. So the next step is to fix #6.

**Comment by gavinking** [comment]
> I'm down to 9 failures

Make that 6 failures.
